### DISCUSSION TITLE
Only allow DataFrame as data for DataTable

### DISCUSSION THREAD

**Discussion by iron3oxide** [reflex-dev/reflex#649] [Ideas]
Currently there are multiple ways to give a `DataTable` its `data`. It can be a pandas DataFrame, a list of dictionaries and so forth. This means there has to be a not-so-trivial process of figuring out which type `data` has and then proceeding accordingly, while also checking for conflicts like DataFrame + a columns argument. 

While it is great to offer users the flexibility to define their table data how they see fit, I argue that it would be cleaner and less complex to coerce them to use a "real" tabular data structure, the DataFrame. I also implore you to check out [polars](https://github.com/pola-rs/polars), a faster pandas alternative which may be an even better fit. DataFrames can easily be created from all kinds of data structures and file types, so it shouldn't be a DX burden.

This way, some code (`utils.is_dataframe()`, the `columns` prop and all associated checks) could be deleted and users would have one obvious way to do things. What do you think?

**Comment by SLLau** [comment]
Yes please to supporting POLARS
